Feature#102

ershovd opened the pull request:
- create GroupedChecklistMenu
- change logic in MenuViewModel to allow expand\collapse and changed menu generation
- add UI in MenuPage.xaml
- create static menu generation

Issues to fix:
- UI look'n'feel
- header for static menu (done, flagged, incident). maybe make it expandeable as well?

ershovd commented:
![simulator screen shot 5 2017 17 33 25](https://user-images.githubusercontent.com/3798051/27869306-2f529852-61a8-11e7-9079-fdfdc56f44a9.png)

@danielqualitykitchen  please make suggestion regarding UI of grouping header (in magenta color). I will implement those. also, should I add a header (and collapse\expand capabilities) for last menu items (Done, Flagged etc) or is it better to remove it altogether (no grouping for those items).

j0nscalet commented:
@ershovd Functionality is really good. I like it.

- Bold section headers - no need for the icon; let's keep it simple
- Remove the index in the menu, that's not needed
- Remove the gray separator row between checklist menu items and normal items

j0nscalet commented:
@ershovd I updated my comment and removed request to 'Remove the "extra" section header', I know that's difficult to do because of the way the listview is grouped.

The one bigger thing that is missing and needs to be added is the indicators in the section header(s). See the sketches on the connect issued (#102) for more details.

It should be  simple to implement too based on what's here. A view that is bound to `GroupedChecklistMenuItem.IsExpanded` should do the trick.

ershovd commented:
@j0nscalet I've actually almost did the hiding of the menu header and the other points. 
what do you mean by indicators? the "up" sign (for expanded state) and "down" (for collapsed), is it?

ershovd commented:
@j0nscalet also, do you plan to push other changes to the branch? I am trying to resolve conflicts on it, and would like to know if you are done, so I can proceed

ershovd commented:
@j0nscalet I've added Up and Down icons to checklistHeaderDataTemplate and logic for swtiching them on click. Take a look. Also, I've updated UI according to your comments on GH.
